osl/namespace_linux.go

```diff
@@ -606,9 +606,21 @@ func reexecSetIPv6() {
 		value = byte('0')
 	}
 
-	if err = ioutil.WriteFile(path, []byte{value, '\n'}, 0644); err != nil {
-		logrus.Errorf("failed to %s IPv6 forwarding for container's interface %s: %v", action, os.Args[2], err)
-		os.Exit(4)
+	_, err = os.Stat(path)
+
+	if err != nil {
+		if os.IsNotExist(err) {
+			logrus.Warnf("file does not exist: %s : %v Has IPv6 been disabled in this node's kernel?", path, err)
+		} else {
+			// We have some serious error, print it out
+			logrus.Errorf("failed to stat %s : %v", path, err)
+			os.Exit(4)
+		}
+	} else {
```

Comment:
Could probably get rid of the `else` in that case (as we already exit). If we're ignoring the non-existence the the file, the appropriate exit code would be `0` (non-error)

```go
if _, err := os.Stat(path); err != nil {
	if os.IsNotExist(err) {
		os.Exit(0)
	}
	logrus.Errorf("failed to stat %s : %v", path, err)
	os.Exit(4)
}

if err = ioutil.WriteFile(path, []byte{value, '\n'}, 0644); err != nil {
	logrus.Errorf("failed to %s IPv6 forwarding for container's interface %s: %v", action, os.Args[2], err)
	os.Exit(4)
}

os.Exit(0)
```

Reply:
Good idea, thanks!

I'm still inclined to include the warning if the file doesn't exist, though.